### DISCUSSION TITLE
Provide Nuget packages for avalonia essentials and basic instruction to get it working

### DISCUSSION THREAD

**Discussion by CastelloBrancoTecnologia** [AvaloniaUI/Avalonia#12891] [General]
In documentation (dealing with multiplataform) avalonia mention avalonia.essentials, but there arent nuget packages nor instructions to get it working

Also Avalonia studio dont have option to use avalonia essentials, please put it in avalonia studio too 

Last, in dealing multiplataform there arent examples of thecniques described.

**Comment by timunie** [comment]
It's a WIP project for now. You can only consume it as project reference for the time beeing. Once it's production ready from our perspective, a nuget will be released. 

if you're talking abut https://github.com/VitalElement/AvalonStudio, I'm not sure what the plans in long term are. @danwalmsley may know more. If you are looking for a great IDE developing with Avalonia, I can say Rider has great support. Recently we also got an extension to VSCode which you can use. 

----

I consider this a general question and not a bug, so I'll convert it into a discussion for now.

**Reply by cesarchefinho** [reply]
It is bug because you have an entire section in documentation (dealining multiplataform) and there arent packages and instruction to use it (the bug is in documentatio)

also it is an enhancement bug.

For Avalonia stuidio it is part of project template, there are option when you create a project in visual studio like use mmvm, compiled bindings etc. we need avalonia essentials in these options.

**Reply by timunie** [reply]
Having a project mentioned in the docs doesn't mean it need to provide a recent nuget (yet), but adding a warning in the docs could and should be an option. In that case the issue (or a PR) goes into docs repo. 

I didn't notice your are talking about template studio, sorry for that. As I personally use Rider, I have no up-to-date info what can be pulled in. But there are so many options. like Essentials, Behaviors, ColorPicker, DataGrid,... Don't know if we can list all of them there. I'll ask the team about this particular question.

**Reply by timunie** [reply]
Avalonia.Essentials will be removed from the docs for now until they are production ready.

**Comment by vpenades** [comment]
In the meantime, I've been trying to use Microsoft.Maui.Essentials as a temporary replacement, and it's quite painful.

It seems Maui on windows leverages the UWP APIs , so many APIs simply don't work on Win32 desktop applications regardless of existing perfectly valid and usable APIs.

Avalonia.Essentials is very much needed.

**Reply by stevemonaco** [reply]
There's nothing wrong with using UWP/WinRT APIs. They exist to be used. Win10, which supports them, is going end-of-life in less than half a year.

If you're supporting OS's that are older than that (no longer officially supported by Microsoft, .NET, or Avalonia), you can expect a degraded experience. It might be personally important, but nobody wants to waste tremendous amounts of time backporting features to old OS's.

**Comment by cedric56** [comment]
@vpenades Here is a port of Microsoft.Maui.Essentials for avalonia
https://www.nuget.org/packages/Avae.Windows.Essentials/
